File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        int[] answer = new int[2];
        
        List<Integer> dq = new ArrayList<>();
        
        for(int i = 0; i < operations.length; i++){
            String[] s = operations[i].split(" ");
            if(s[0].equals("I")){
                dq.add(Integer.parseInt(s[1]));
            }
            else if(!dq.isEmpty() && s[0].equals("D")){
                if(s[1].equals("1"))
                    dq.remove(0);
                else
                    dq.remove(dq.size() - 1);
            }
            dq.sort((o1, o2) -> (o2 - o1));
        }
        
        if(dq.isEmpty()){
            answer[0] = 0;
            answer[1] = 0;
        }else{
            answer[0] = dq.get(0);
            answer[1] = dq.get(dq.size() - 1);
        }
        
        return answer;
    }
}

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    answer = []
    
    q = []
    
    for o in operations:
        a, n = o.split(" ")
        if a == "I":
            q.append(int(n))
        else:
            if len(q) > 0:
                if n == "1":
                    q.remove(max(q))
                else:
                    q.remove(min(q))
    
    if len(q) == 0:
        answer = [0, 0]
    else:
        answer.append(max(q))
        answer.append(min(q))
    
    return answer